File: shared-atlas-dart/lib/src/interfaces/mobile_auth_response.dart
import 'package:shared_atlas_dart/shared_atlas_dart.dart';

/// Prefix: Mobile
/// Used for mobile app authentication response.
class MobileAuthResponse {
  final IUser user;
  final dynamic tokens;

  MobileAuthResponse({
    required this.user,
    required this.tokens,
  });

  factory MobileAuthResponse.fromJson(Map<String, dynamic> json) {
    return MobileAuthResponse(
      user: IUser.fromJson(json['user'] as Map<String, dynamic>),
      tokens: json['tokens'] as dynamic,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'user': this.user.toJson(),
      'tokens': this.tokens,
    };
  }
}

File: packages/shared-atlas-rust/src/interfaces/mobile_auth_response.rs
use serde::{Serialize, Deserialize};
use crate::interfaces::iuser::IUser;
use serde_json;

/// Prefix: Mobile
/// Used for mobile app authentication response.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MobileAuthResponse {
    pub user: IUser,
    pub tokens: serde_json::Value,
}